### PR TITLE
feat: Asynchronous connection to mqtt  broker

### PR DESCRIPTION
**Pull request by owny990312**

### CODE REVIEW

File: src/plugins/mqtt_plugin/mqtt_plugin.cc [hunk 1]
```diff
@@ -71,42 +79,8 @@ bool MqttPlugin::Initialize(runtime::core::AimRTCore *core_ptr) noexcept {
...
-      static_cast<std::promise<bool> *>(context)->set_value(false);
-    };
-    conn_opts.context = &connect_ret_promise;
-    int rc = MQTTAsync_connect(client_, &conn_opts);
-    AIMRT_CHECK_ERROR_THROW(rc == MQTTASYNC_SUCCESS, "Failed to connect mqtt broker, return code: {}", rc);
-
-    bool connect_ret = connect_ret_promise.get_future().get();
-    AIMRT_CHECK_ERROR_THROW(connect_ret, "Failed to connect mqtt broker");
+    // connect to broker, which is an async operation
+    Connect();
```
Comment (zhangyi1357) [inline src/plugins/mqtt_plugin/mqtt_plugin.cc:83]: Named as AsyncConnect or AsyncConnectToBroker might be better?

File: src/plugins/mqtt_plugin/mqtt_plugin.h [hunk 2]
```diff
@@ -22,6 +22,9 @@ class MqttPlugin : public AimRTCorePluginBase {
     std::string client_id;
     uint32_t max_pkg_size_k = 1024;
     std::string truststore;
+    std::string client_cert;
+    std::string client_key;
+    std::string client_key_password;
```
Comment (AimRT) [inline src/plugins/mqtt_plugin/mqtt_plugin.h:27]: Doc?

File: document/sphinx-cn/tutorials/plugins/mqtt_plugin.md [hunk 3]
```diff
@@ -23,13 +23,16 @@
 | client_id      | string | 必选     | ""     | 本节点的 mqtt client id |
 | max_pkg_size_k | int    | 可选     | 1024   | 最大包尺寸，单位：KB    |
 | truststore     | string | 可选     | ""     | broker的CA证书路径      |
-
+| client_cert    | string | 可选     | ""     | 客户端证书路径          |
+| client_key     | string | 可选     | ""     | 客户端私钥路径          |
```
Comment (AimRT) [inline document/sphinx-cn/tutorials/plugins/mqtt_plugin.md:27]: Add an example for these options

File: src/plugins/mqtt_plugin/mqtt_plugin.cc [hunk 4]
```diff
@@ -71,42 +83,8 @@ bool MqttPlugin::Initialize(runtime::core::AimRTCore *core_ptr) noexcept {
...
-      static_cast<std::promise<bool> *>(context)->set_value(false);
-    };
-    conn_opts.context = &connect_ret_promise;
-    int rc = MQTTAsync_connect(client_, &conn_opts);
-    AIMRT_CHECK_ERROR_THROW(rc == MQTTASYNC_SUCCESS, "Failed to connect mqtt broker, return code: {}", rc);
-
-    bool connect_ret = connect_ret_promise.get_future().get();
-    AIMRT_CHECK_ERROR_THROW(connect_ret, "Failed to connect mqtt broker");
+    // connect to broker, which is an async operation
+    AsyncConnect();
```
Comment (AimRT) [inline src/plugins/mqtt_plugin/mqtt_plugin.cc:87]: We can wait seconds after this line

File: src/examples/plugins/mqtt_plugin/README.md [hunk 5]
```diff
@@ -133,3 +133,81 @@
...
+
+  # 生成客户端证书签名请求 (CSR)
+  openssl req -new -key client_key.pem -out client_csr.pem
+
+  # 使用 CA 签署客户端证书
+  openssl x509 -req -in client_csr.pem -CA ca_crt.pem -CAkey ca_key.pem -CAcreateserial -out client_crt.pem -days 365 -sha256
+  ```
+
+
+  ````shell
```
Comment (AimRT) [inline src/examples/plugins/mqtt_plugin/README.md:193]: Looks like an extra  `

Comment (zhangyi1357) [inline src/examples/plugins/mqtt_plugin/README.md:193]: Looks like there's an extra `

File: src/examples/plugins/mqtt_plugin/install/linux/bin/cfg/examples_plugins_mqtt_plugin_pb_chn_pub_with_ssl_cfg.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+aimrt:
+  plugin:
+    plugins:
+      - name: mqtt_plugin
+        path: ./libaimrt_mqtt_plugin.so
+        options:
+          broker_addr: ssl://127.0.0.1:8883
+          client_id: example_pb_chn_publisher
+          max_pkg_size_k: 1024
+          truststore: /etc/emqx/certs/cacert.pem
```
Comment (AimRT) [inline src/examples/plugins/mqtt_plugin/install/linux/bin/cfg/examples_plugins_mqtt_plugin_pb_chn_pub_with_ssl_cfg.yaml:13]: Don't rely on the environment

**Pull request merged**